tree/dataframe/src/RDFInterfaceUtils.cxx

```diff
@@ -703,14 +711,26 @@ std::shared_ptr<RNodeBase> UpcastNode(std::shared_ptr<RNodeBase> ptr)
 ColumnNames_t GetValidatedColumnNames(RLoopManager &lm, const unsigned int nColumns, const ColumnNames_t &columns,
                                       const ColumnNames_t &validDefines, RDataSource *ds)
 {
-   const auto &defaultColumns = lm.GetDefaultColumnNames();
-   auto selectedColumns = SelectColumns(nColumns, columns, defaultColumns);
-   const auto &validBranchNames = lm.GetBranchNames();
-   const auto unknownColumns =
-      FindUnknownColumns(selectedColumns, validBranchNames, validDefines, ds ? ds->GetColumnNames() : ColumnNames_t{});
+   auto selectedColumns = SelectColumns(nColumns, columns, lm.GetDefaultColumnNames());
+
+   // Resolve aliases and expand `#var` to `__rdf_sizeof_var`
+   const auto &aliasMap = lm.GetAliasMap();
+   const auto aliasMapEnd = aliasMap.end();
+
+   for (auto idx : ROOT::TSeqU(selectedColumns.size())) {
+      auto &colName = selectedColumns[idx];
+      const auto aliasColumnNameIt = aliasMap.find(colName);
+      if (aliasMapEnd != aliasColumnNameIt)
+         colName = aliasColumnNameIt->second;
+      if (colName[0] == '#')
+         colName = "__rdf_sizeof_" + colName.substr(1);
```

Comment:
if `colName == "#"`, this will throw an out of range exception (which may or may not be desired).

Reply:
fixed